apps/meteor/ee/server/local-services/omnichannel/OmniEEService.ts

```diff
@@ -0,0 +1,175 @@
+import type { Job } from '@rocket.chat/agenda';
+import type { IOmnichannelRoom, IRoom, IUser } from '@rocket.chat/core-typings';
+import { LivechatRooms, Users } from '@rocket.chat/models';
+import moment from 'moment';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { Livechat } from '../../../../app/livechat/server';
+import { forwardRoomToAgent } from '../../../../app/livechat/server/lib/Helper';
+import { RoutingManager } from '../../../../app/livechat/server/lib/RoutingManager';
+import { settings } from '../../../../app/settings/server';
+import { ServiceClassInternal } from '../../../../server/sdk/types/ServiceClass';
+import { schedulerLogger } from '../../../app/livechat-enterprise/server/lib/logger';
+import { OmniJobSchedularService } from '../../sdk';
+import type { IOmniEEService } from '../../sdk/types/IOmniEEService';
+import { OMNI_JOB_NAME } from './OmniJobSchedularService';
+
+type OmniOnHoldJobData = {
+	roomId: IRoom['_id'];
+	comment: string;
+};
+
+type OmniAutoTransferUnansweredChatJobData = {
+	roomId: IRoom['_id'];
+};
+
+export class OmniEEService extends ServiceClassInternal implements IOmniEEService {
+	protected name = 'omni-ee-service';
+
+	private logger = schedulerLogger;
+
+	private systemUser: IUser;
+
+	constructor() {
+		super();
+		this.logger.debug('OmniEEService started');
+	}
+
+	async monitorOnHoldRoomForAutoClose(roomId: string, timeout: number, comment: string): Promise<void> {
```

Comment:
What if we have a "sub-service" for holding this logic? And then on constructor of this service, we initialize it. 
So, you can use those things in the form of "OmniEEService.scheduling.monitorxxxx" 🤔 

(It just passed my mind, but, would be better to keep things "separated")

Reply:
I like this idea and I tried to implement it here. Overall it went good but there is just a small issue with SDK due to which the methods using the new service SDK have to apply 2 promises like this. 
`await (await OmniEEService.getSchedulingSubService()).monitorUnansweredRoomForAutoTransfer(rid, autoTransferTimeout);`

One promise to get the inner class & then the other promise to get actually execute the method. This is happening because the proxy SDK used for services is always async - so all the methods contained within the service will be async.

I also tried to assign a public variable so we can access it as you mentioned above - `OmniEEService.scheduling.monitorxxx`  - however even this isn't working & it also requires an async operation.

@KevLehman  Let me know if you have any suggestions on how to improve this or if you'd rather prefer to revert this change & avoid sub-service in order to avoid multiple promises

Reply:
Yeah, probably that's because 🤔 of how it's being done. 
What if we change scheduling service to serve purely as scheduling? So instead of a "sub service", create a separate service for "scheduling" things, without any extra logic, just agenda interaction. Like

```
ScheduleService.scheduleAction('OmniEEfdafddafd', () => {});
```

So, it's called from inside the service (OmniEEService) and all the Omni-logic for scheduling is managed there 👀 . That way,  you avoid the await await thing 👀 wdyt?

Reply:
Isn't this what `OmniJobSchedulerService` does? :thinking:  It only has interactions with agenda lib & `OmniEEService` uses it to just schedule jobs

Reply:
JFYI: I've moved back the sub-service-related logic :leftwards_arrow_with_hook:

Reply:
If that's what it does, I'll suggest renaming the service 👀 so it's generic and can be used in other places
Instead of "OmniJob..." just "Scheduler" so ppl don't think it's omni specifici